plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -142,41 +165,157 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 			}
 		}
 
-		// If this element is the LCP (for a breakpoint group), add a preload link for it.
-		foreach ( $context->url_metric_group_collection->get_groups_by_lcp_element( $xpath ) as $group ) {
-			$link_attributes = array_merge(
+		$parent_tag = $this->get_parent_tag_name( $context );
+		if ( 'PICTURE' !== $parent_tag ) {
+			$this->add_image_preload_link_for_lcp_element_groups(
+				$context,
+				$xpath,
 				array(
-					'rel'           => 'preload',
-					'fetchpriority' => 'high',
-					'as'            => 'image',
-				),
-				array_filter(
-					array(
-						'href'        => (string) $processor->get_attribute( 'src' ),
-						'imagesrcset' => (string) $processor->get_attribute( 'srcset' ),
-						'imagesizes'  => (string) $processor->get_attribute( 'sizes' ),
-					),
-					static function ( string $value ): bool {
-						return '' !== $value;
-					}
+					'href'        => $processor->get_attribute( 'src' ),
+					'imagesrcset' => $processor->get_attribute( 'srcset' ),
+					'imagesizes'  => $processor->get_attribute( 'sizes' ),
+					'crossorigin' => $this->get_attribute_value( $processor, 'crossorigin' ),
 				)
 			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Process a <picture> element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_HTML_Tag_Processor  $processor HTML tag processor.
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context.
+	 *
+	 * @return bool Whether the tag should be tracked in URL Metrics.
+	 */
+	private function process_picture( OD_HTML_Tag_Processor $processor, OD_Tag_Visitor_Context $context ): bool {
+		$collected_sources = array();
+		$img_xpath         = null;
+
+		// Loop through child tags until we reach the closing </picture> tag.
+		while ( $processor->next_tag() ) {
+			$tag = $processor->get_tag();
+
+			// If we reached the closing </picture> tag, break.
+			if ( 'PICTURE' === $tag && $processor->is_tag_closer() ) {
+				break;
+			}
+
+			// Collect <source> elements.
+			if ( 'SOURCE' === $tag && ! $processor->is_tag_closer() ) {
+				$media = $processor->get_attribute( 'media' );
+				$type  = $processor->get_attribute( 'type' );
 
-			$crossorigin = $this->get_attribute_value( $processor, 'crossorigin' );
-			if ( null !== $crossorigin ) {
-				$link_attributes['crossorigin'] = 'use-credentials' === $crossorigin ? 'use-credentials' : 'anonymous';
+				// Ensure that all <source> elements have a type attribute and no media attribute.
+				if ( null !== $media || null === $type ) {
+					return false;
+				}
+
+				$collected_sources[] = array(
+					'srcset'      => $processor->get_attribute( 'srcset' ),
+					'sizes'       => $processor->get_attribute( 'sizes' ),
+					'type'        => $type,
+					'crossorigin' => $this->get_attribute_value( $processor, 'crossorigin' ),
```

Comment:
Actually, `crossorigin` is only defined on the `IMG`. 

Also, I realized that we neglected `referrerpolicy`.